### ISSUE TITLE
[Spring] Required post body with path parameters produces uncompilable code

### ISSUE THREAD

**Issue by staale** [swagger-api/swagger-codegen-generators#255]
If you have a required requestBody in combination with path parameters, the body will be placed first, and there will be no comma after it in the generated codes argument list:

Attached sample openapi.yaml, invoke with:

```
java -jar swagger-codegen-cli-3.0.3.jar generate -i openapi.yaml -l spring -o test
```

Resulting code:
```
    ResponseEntity<Body> failure(
        @ApiParam(value = "" ,required=true )  @Valid @RequestBody
        Body body@ApiParam(value = "First value",required=true) @PathVariable("first") 
        String first,
        @ApiParam(value = "Second value",required=true) @PathVariable("second") String second);
```
I have reformatted the code slightly to highlight the error.

[openapi.txt](https://github.com/swagger-api/swagger-codegen-generators/files/2657452/openapi.txt)

**Comment by totothekiller** [comment]
I have the same issue (with swagger-codegen-maven-plugin 3.0.3)
In my case, the type of @RequestBody body variable is wrongly set to Object.

**Comment by voydz** [comment]
Hi,

Maybe **this issue has to be generalized?** I am experiencing the same issue for both csharp generators (csharp and csharp-dotnet2).

**Example:**

Request method definition:
```json
...
"/api/v1/Device/AddUserDevice":{
   "post":{
      "tags":[
         "Device"
      ],
      "operationId":"Device_AddUserDevice",
      "consumes":[
         "application/json",
         "text/json",
         "application/xml",
         "text/xml",
         "application/x-www-form-urlencoded"
      ],
      "produces":[
         "application/json",
         "text/json",
         "application/xml",
         "text/xml"
      ],
      "parameters":[
         {
            "name":"value",
            "in":"body",
            "required":true,
            "schema":{
               "$ref":"#/definitions/DeviceModel"
            }
         }
      ],
      "responses":{
         "200":{
            "description":"OK"
         },
         "400":{
            "description":"BadRequest"
         }
      }
   }
}
...
```

Body schema definition:
```json
...
"DeviceModel":{
   "type":"object",
   "properties":{
      "id":{
         "type":"string"
      },
      "created":{
         "format":"date-time",
         "type":"string"
      },
      "additionalInformation":{
         "type":"string"
      },
      "deviceId":{
         "type":"string"
      },
      "deviceTypeId":{
         "format":"int32",
         "type":"integer"
      },
      "userId":{
         "type":"string"
      },
      "deviceName":{
         "type":"string"
      },
      "controlAuthorizationLevel":{
         "format":"int32",
         "type":"integer"
      },
      "isOwnedByMe":{
         "type":"boolean"
      },
      "isSharedToOthers":{
         "type":"boolean"
      }
   }
}
...
```

And the output I get for `<SwaggerGeneratedProjectRoot>/Api/DeviceApi.cs` is:
```csharp
    /// <summary>
    /// Represents a collection of functions to interact with the API endpoints
    /// </summary>
    public interface IDeviceApi : IApiAccessor
    {
        #region Synchronous Operations
        /// <summary>
        /// 
        /// </summary>
        /// <remarks>
        /// 
        /// </remarks>
        /// <exception cref="Api.Client.ApiException">Thrown when fails to make API call</exception>
        /// <param name="body"></param>
        /// <param name="id"> (optional)</param>
        /// <param name="created"> (optional)</param>
        /// <param name="additionalInformation"> (optional)</param>
        /// <param name="deviceId"> (optional)</param>
        /// <param name="deviceTypeId"> (optional)</param>
        /// <param name="userId"> (optional)</param>
        /// <param name="deviceName"> (optional)</param>
        /// <param name="controlAuthorizationLevel"> (optional)</param>
        /// <param name="isOwnedByMe"> (optional)</param>
        /// <param name="isSharedToOthers"> (optional)</param>
        /// <returns></returns>
        void DeviceAddUserDevice (DeviceModel bodystring id = null, DateTime? created = null, string additionalInformation = null, string deviceId = null, int? deviceTypeId = null, string userId = null, string deviceName = null, int? controlAuthorizationLevel = null, bool? isOwnedByMe = null, bool? isSharedToOthers = null);
...
```

**Comment by wvitte** [comment]
Strangely it works for me if I removed, 

` "required":true,`

**Comment by HugoMario** [comment]
hello guys,  this issue has been fixed with PR #267. Going to close this ticket but please let me know if there's still something wrong and i'll handle it.
